File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        multiset<int> st1(begin(nums1),end(nums1));
        multiset<int> st2(begin(nums2),end(nums2));
        vector<int> ans;
        set_intersection(begin(st1),end(st1),begin(st2),end(st2),back_inserter(ans));
        return ans;
    }
};

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        inter = set(nums1) & set(nums2)
        res = []
        for n in inter:
            res.extend([n] * nums1.count(n) if nums1.count(n) <= nums2.count(n) else [n] * nums2.count(n))
        return res    